doc/machine-auto.rst

```diff
@@ -0,0 +1,33 @@
+machine_dict: 
+    | type: ``dict``
+    | argument path: ``machine_dict``
+
+    batch_type: 
+        | type: ``str``
+        | argument path: ``machine_dict/batch_type``
+
+        The batch job system type. Option: Slurm, PBS, LSF, Shell, DpCloudServer
+
+    context_type: 
+        | type: ``str``
+        | argument path: ``machine_dict/context_type``
+
+        The connection used to remote machine. Option: LocalContext, LazyLocalContext, SSHContext， DpCloudServerContext
```

Comment:
similar as above. Also having to write "Context" in the choice is redundant. I can help do a refactor and solve this.